ext/node/polyfills/_http_outgoing.ts

```diff
@@ -584,6 +576,170 @@ export class OutgoingMessage extends Stream {
     return headers;
   }
 
+  _storeHeader(firstLine: string, _headers: never) {
+    // firstLine in the case of request is: 'GET /index.html HTTP/1.1\r\n'
+    // in the case of response it is: 'HTTP/1.1 200 OK\r\n'
+    const state = {
+      connection: false,
+      contLen: false,
+      te: false,
+      date: false,
+      expect: false,
+      trailer: false,
+      header: firstLine,
+    };
+
+    const headers = this[kOutHeaders];
+    if (headers) {
+      // headers is null-prototype object, so ignore the guard lint
+      // deno-lint-ignore guard-for-in
+      for (const key in headers) {
+        const entry = headers[key];
+        this._matchHeader(state, entry[0], entry[1]);
+      }
+    }
+
+    // Date header
+    if (this.sendDate && !state.date) {
+      this.setHeader("Date", utcDate());
+    }
+
+    // Force the connection to close when the response is a 204 No Content or
+    // a 304 Not Modified and the user has set a "Transfer-Encoding: chunked"
+    // header.
+    //
+    // RFC 2616 mandates that 204 and 304 responses MUST NOT have a body but
+    // node.js used to send out a zero chunk anyway to accommodate clients
+    // that don't have special handling for those responses.
+    //
+    // It was pointed out that this might confuse reverse proxies to the point
+    // of creating security liabilities, so suppress the zero chunk and force
+    // the connection to close.
+    if (
+      this.chunkedEncoding && (this.statusCode === 204 ||
+        this.statusCode === 304)
+    ) {
+      debug(
+        this.statusCode + " response should not use chunked encoding," +
+          " closing connection.",
+      );
+      this.chunkedEncoding = false;
+      this.shouldKeepAlive = false;
+    }
+
+    // TODO(osddeitf): this depends on agent and underlying socket
+    // keep-alive logic
+    // if (this._removedConnection) {
```

Comment:
Why is this commented out? It seems you changed the code to call `this.send()` which appears to already by using connection directly so this part should work as well

Reply:
That part is hard to understand, so let me elaborate (sorry if it is too long):
- The current implementation of `ClientRequest` is not using any `Agent`, there are some comments in `ClientRequest` constructor:
    ```typescript
    // initiate connection
    // TODO(crowlKats): finish this
    /*if (this.agent) {
      this.agent.addRequest(this, optsWithoutSignal);
    ```
    In Deno, the `node:http` requests are sent by calling `core.ops.op_node_http_request`, which is in turn make a new `reqwest`, not using any `Agent`.
    It's different from what I see in Nodejs implementation of `node:http` is that the `Agent` is used to manage the underlying socket connection (TCP), and the keepalive logics is to reuse the existing connections when possible.
- The logics I commented out is to add headers `Connection` as `close` or `keep-alive`, which is responsible for keepalive logics.
- The variables `this._last` and `this.shouldKeepAlive` are modified as well. They are used in the following lines in Nodejs implementation, to close the sockets:
    - https://github.com/nodejs/node/blob/v18.x/lib/_http_agent.js#L159-L166
    - https://github.com/nodejs/node/blob/v18.x/lib/_http_server.js#L972-L977
- With the crate `reqwest`, there's no way to access the underlying socket connections, as far as I know (`reqwest` may has its own way to reuse connections). So effectively, we cannot implement keepalive logics in Deno `ClientRequest`, I commented those lines to prevent HTTP header `Connection` being added, to avoid conflict with how the sockets are reused by `reqwest` crate.

Reply:
Thanks, that makes sense to me. I plan to rewrite both `fetch` and `node:http` implementations to not use `reqwest`, but use `hyper` directly. 

@crowlKats can you take a look here? If it's fine for you too we can land it.

Reply:
It's been almost a week. How's it going?